src/System.Linq.Expressions/src/System/Linq/Expressions/MethodCallExpression.cs

```diff
@@ -212,6 +223,40 @@ internal override MethodCallExpression Rewrite(Expression instance, IList<Expres
         }
     }
 
+    internal class MethodCallExpression0 : MethodCallExpression, IArgumentProvider
```

Comment:
We definitely can. There are no derived types. Haven't seen many other internal types that are sealed in this project though, but I wouldn't mind.

Sent from my Windows Phone
________________________________
From: Stephen Toub<mailto:notifications@github.com>
Sent: ‎9/‎17/‎2015 3:53 PM
To: dotnet/corefx<mailto:corefx@noreply.github.com>
Cc: Bart De Smet<mailto:bartde@microsoft.com>
Subject: Re: [corefx] Use optimized N-ary MethodCallExpression nodes from all Call factories (#3288)


In src/System.Linq.Expressions/src/System/Linq/Expressions/MethodCallExpression.cs<https://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fdotnet%2fcorefx%2fpull%2f3288%23discussion_r39808619&data=01%7c01%7cbartde%40microsoft.com%7c30c80a7ce692465e76d708d2bfb2ceb9%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=bpfB9yXjelvEfG%2b2CUqMVH3IeTy7dfLozEezUZCcmSQ%3d>:

> @@ -212,6 +223,40 @@ internal override MethodCallExpression Rewrite(Expression instance, IList<Expres
>          }
>      }
>
> +    internal class MethodCallExpression0 : MethodCallExpression, IArgumentProvider


Can we seal types like this? Or do we not do that in this library? (Or maybe something somewhere derives from this?)

—
Reply to this email directly or view it on GitHub<https://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fdotnet%2fcorefx%2fpull%2f3288%2ffiles%23r39808619&data=01%7c01%7cbartde%40microsoft.com%7c30c80a7ce692465e76d708d2bfb2ceb9%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=fLo3nU%2fjEcs2FI%2fQt51bSqfX%2fRwUW1wk65AbN3YvbeU%3d>.